test/hotspot/jtreg/containers/cgroup/NestedCgroup.java

```diff
@@ -0,0 +1,222 @@
+            args.add(jdkTool);
+            args.add("-cp");
+            args.add(System.getProperty("test.classes"));
+        }
+
+        public static void args_add_self_verbose(List<String> args) {
+            args_add_self(args);
+            args.add("-XshowSettings:system");
+            args.add("-Xlog:os+container=trace");
+        }
+
+        public Test(String controller_) throws Exception {
+            controller = controller_;
+            OutputAnalyzer output;
+
+            List<String> cgdelete = new ArrayList<>();
+            cgdelete.add("cgdelete");
+            cgdelete.add("-r");
+            cgdelete.add("-g");
+            cgdelete.add(controller + ":" + CGROUP_OUTER);
+            output = ProcessTools.executeProcess(new ProcessBuilder(cgdelete));
+            if (output.contains("cgdelete: libcgroup initialization failed: Cgroup is not mounted")) {
+                throw new SkippedException("cgroup/cgroup2 is not mounted: " + output.getStderr());
+            }
+            output.stdoutShouldBeEmpty();
+            if (!output.contains("cgdelete: cannot remove group '" + CGROUP_OUTER + "': No such file or directory")) {
+                output.stderrShouldBeEmpty();
+            }
+
+            // Alpine Linux 3.20.1 needs cgcreate1 otherwise:
+            // cgcreate: can't create cgroup [...]: No such file or directory
+            List<String> cgcreate1 = new ArrayList<>();
+            cgcreate1.add("cgcreate");
+            cgcreate1.add("-g");
+            cgcreate1.add(controller + ":" + CGROUP_OUTER);
+            output = ProcessTools.executeProcess(new ProcessBuilder(cgcreate1));
+            if (output.contains("cgcreate: can't create cgroup " + CGROUP_OUTER + ": Cgroup, operation not allowed")
+             || output.contains("cgcreate: can't create cgroup " + CGROUP_OUTER + ": Cgroup, requested group parameter does not exist")) {
+                throw new SkippedException("Missing root permission: " + output.getStderr());
+            }
+            output.stdoutShouldBeEmpty();
+            output.stderrShouldBeEmpty();
+
+            List<String> cgcreate2 = new ArrayList<>();
+            cgcreate2.add("cgcreate");
+            cgcreate2.add("-g");
+            cgcreate2.add(controller + ":" + CGROUP_OUTER + "/" + CGROUP_INNER);
+            output = ProcessTools.executeProcess(new ProcessBuilder(cgcreate2));
+            output.stdoutShouldBeEmpty();
+            output.stderrShouldBeEmpty();
+
+            String provider = Metrics.systemMetrics().getProvider();
+            System.err.println("Metrics.systemMetrics().getProvider() = " + provider);
+            boolean isCgroup2;
+            if ("cgroupv1".equals(provider)) {
+              isCgroup2 = false;
+            } else if ("cgroupv2".equals(provider)) {
+              isCgroup2 = true;
+            } else {
+              throw new IllegalArgumentException();
+            }
```

Comment:
It cannot because jdk.internal.platform.Metrics is JDK-private and it must be exported by jtreg `@modules java.base/jdk.internal.platform`. And this jtreg export does not work for the newly spawned Java as I cannot replicate the jtreg commandline, I haven't found it in system properties.